src/app/hooks/useViewTracker/index.test.jsx

```diff
@@ -0,0 +1,253 @@
+    setIntersectionNotObserved();
+    rerender();
+
+    act(() => jest.advanceTimersByTime(1000));
+
+    expect(global.fetch).not.toHaveBeenCalled();
+  });
+
+  it('should not send event to ATI more than once when element is scrolled in and out of view', async () => {
+    setIntersectionNotObserved();
+
+    const trackingData = {
+      pageData,
+      componentName: 'mostRead',
+      actionLabel: 'most-read-view',
+    };
+    const { rerender } = renderHook(() => useViewTracker(trackingData), {
+      wrapper,
+    });
+
+    // scroll element into view
+    setIntersectionObserved();
+    rerender();
+    act(() => jest.advanceTimersByTime(1100));
+
+    // scroll element out of view
+    setIntersectionNotObserved();
+    rerender();
+
+    // scroll element into view again
+    setIntersectionObserved();
+    rerender();
+    act(() => jest.advanceTimersByTime(1100));
+
+    const [[viewEventUrl]] = global.fetch.mock.calls;
+
+    expect(viewEventUrl).toMatch(process.env.SIMORGH_ATI_BASE_URL);
+    expect(global.fetch).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('Error handling', () => {
+  it('should not throw error and not send event to ATI when no tracking data passed into hook', async () => {
+    setIntersectionObserved();
+
+    const trackingData = undefined;
+
+    const { result } = renderHook(() => useViewTracker(trackingData), {
+      wrapper,
+    });
+
+    act(() => jest.advanceTimersByTime(1100));
+
+    expect(result.error).toBeUndefined();
+    expect(global.fetch).not.toHaveBeenCalled();
+  });
+
+  it('should not throw error and not send event to ATI when no pageData passed into hook', async () => {
+    setIntersectionObserved();
+
+    const trackingData = {
+      pageData: {},
+      componentName: 'mostRead',
+      actionLabel: 'most-read-view',
+    };
+
+    const { result } = renderHook(() => useViewTracker(trackingData), {
+      wrapper,
+    });
+
+    act(() => jest.advanceTimersByTime(1100));
+
+    expect(result.error).toBeUndefined();
+    expect(global.fetch).not.toHaveBeenCalled();
+  });
+
+  it('should not throw error and not send event to ATI when unexpected data passed into hook', async () => {
+    setIntersectionObserved();
+
+    const trackingData = {
+      foo: 'bar',
+    };
+
+    const { result } = renderHook(() => useViewTracker(trackingData), {
+      wrapper,
+    });
+
+    act(() => jest.advanceTimersByTime(1100));
+
+    expect(result.error).toBeUndefined();
+    expect(global.fetch).not.toHaveBeenCalled();
+  });
+
+  it('should not throw error and not send event to ATI when unexpected data type passed into hook', async () => {
+    setIntersectionObserved();
+
+    const trackingData = ['unexpected data type'];
+
+    const { result } = renderHook(() => useViewTracker(trackingData), {
+      wrapper,
+    });
+
+    act(() => jest.advanceTimersByTime(1100));
+
+    expect(result.error).toBeUndefined();
+    expect(global.fetch).not.toHaveBeenCalled();
+  });
+});
```

Comment:
Just looking through the tests I'd expect there to be a small dummy element being observed like `<div>Element to observe</div>`. Just found the tests difficult to read but they look very comprehensive!

Reply:
we're actually just testing the hook and not the hook being used on an element so there is no element. the hook just returns a ref that can be attached to an element using the ref attribute/prop

Reply:
Makes sense thanks, is there anywhere it might be tested with a real element or would you think that is unecessary/excessive?

Reply:
In whichever components we add this to we can should have a unit test to check the ref returned from the hook is attached to the component's relevant DOM element.